File: go/70.climbing-stairs.go
/*
 * @lc app=leetcode id=70 lang=golang
 *
 * [70] Climbing Stairs
 */
func climbStairs(n int) int {
	// F(n) = F(n-1) + F(n-2)
	results := []int{0, 1, 2}
	if n < 3 {
		return results[n]
	} else {
		for i := 3; i <= n; i++ {
			results = append(results, results[i-1] + results[i-2])
		}
		return results[n]
	}
}

File: csharp/70.climbing-stairs.cs
/*
 * @lc app=leetcode id=70 lang=csharp
 *
 * [70] Climbing Stairs
 */

// @lc code=start
public class Solution 
{
    public int ClimbStairs(int n) 
    {
        int[] dp = new int[n+1];

        dp[0] = 1;
        dp[1] = 1;
        for (var i = 2; i <= n; i++)
        {
            dp[i] = dp[i-1] + dp[i-2];
        }

        return dp[n];
    }
}
// @lc code=end

